src/program/error/syntax.rs

```diff
@@ -0,0 +1,49 @@
+use std::{error::Error, fmt};
+
+use crate::parser::{LexError, ParseError};
+
+/// Failed to parse a Quil program for some reason.
+///
+/// # Display
+///
+/// The standard [`Display`](fmt::Display) implementation shows only the initial error.
+/// Use the alternative syntax (e.g., `format!("{:#}", err)` instead of `format!("{}", err)`) to
+/// see a backtrace of errors that caused this one.
+///
+/// See also [`Error`](crate::parser::Error).
+#[derive(Debug, PartialEq)]
+pub enum SyntaxError {
+    LexError(LexError),
+    ParsingError(ParseError),
+}
```

Comment:
```suggestion
pub enum SyntaxError {
    LexError(LexError),
    ParseError(ParseError),
}
```

Matching tense - `Lexing` and `Parsing` or `Lex` and `Parse`